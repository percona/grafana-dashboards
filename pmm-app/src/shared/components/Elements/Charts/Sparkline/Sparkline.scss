.d3-bar-chart-container .x-axis .domain {
  fill: darkslategray !important;
}
