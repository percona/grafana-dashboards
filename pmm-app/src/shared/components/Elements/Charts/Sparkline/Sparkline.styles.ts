import { css } from 'emotion';

export const styles = {
  graphWrapper: css`
    background: transparent;
    borderBottom: 1px solid grey;
`
};
