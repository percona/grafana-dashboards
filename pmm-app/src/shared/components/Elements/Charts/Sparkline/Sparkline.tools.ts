import * as moment from 'moment';

export const getMetricSparklineKey = (metricName) => {
  switch (metricName) {
    case 'load':
      return 'load';
    case 'num_queries':
      return 'num_queries_per_sec';
    case 'num_queries_with_warnings':
      return 'num_queries_with_warnings_per_sec';
    case 'num_queries_with_errors':
      return 'num_queries_with_errors_per_sec';
    default:
      return `m_${metricName}_sum_per_sec`;
  }
};
export const getAdditionalPoint = (last, previous) => new Date(
  (+moment.utc(last) || 0) - ((+moment.utc(previous) || 0) - (+moment.utc(last) || 0)),
).toISOString();
export const isMetricExists = (metric) => metric === 'NaN' || metric === undefined || metric === '';
export const findYRange = (array, key) => {
  const values = array.map((arrayItem) => +arrayItem[key] || 0);

  return [Math.max(...values) || 1, Math.min(...values) || 0];
};
export const findXRange = (array, key) => {
  const values = array.map((arrayItem) => +moment.utc(arrayItem[key]) || 0);

  return [Math.max(...values), Math.min(...values)];
};
