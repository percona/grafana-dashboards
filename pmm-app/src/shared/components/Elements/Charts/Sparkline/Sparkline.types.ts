export interface PolygonChartInterface {
  data?: any;
  ykey?: any;
  width?: any;
  height?: any;
  metricName?: any;
  color?: any;
  margin?: number;
}
