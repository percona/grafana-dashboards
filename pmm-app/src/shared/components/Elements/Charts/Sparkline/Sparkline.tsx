import React, { useEffect, useRef, useState } from 'react';
import { scaleLinear } from 'd3';
import * as moment from 'moment';
import ReactTooltip from 'react-tooltip';
import { humanize } from '../../../helpers/Humanization';
import { PolygonChartInterface } from './Sparkline.types';
import {
  findXRange,
  findYRange,
  getAdditionalPoint,
  getMetricSparklineKey,
  isMetricExists,
} from './Sparkline.tools';

const updateGraphs = (columnNumber) => {
  const event = new CustomEvent('sync-graphs', { detail: columnNumber });

  document.dispatchEvent(event);
};

const BAR_HEIGHT = 30;
const GRAPH_WIDTH = 300;

export const Sparkline = ({
  margin = 0,
  width = GRAPH_WIDTH,
  height = BAR_HEIGHT,
  metricName,
  data,
  color = '#d0c38b',
}: PolygonChartInterface) => {
  const [tooltip, setTooltip] = useState('');

  const xkey = 'timestamp';
  const ykey = getMetricSparklineKey(metricName);
  let appLoadPolygonChart = [...data] || [];

  if (appLoadPolygonChart.length > 2) {
    appLoadPolygonChart.push({
      point: appLoadPolygonChart[appLoadPolygonChart.length - 1].point + 1,
      timestamp: getAdditionalPoint(
        appLoadPolygonChart[appLoadPolygonChart.length - 1].timestamp,
        appLoadPolygonChart[appLoadPolygonChart.length - 2].timestamp,
      ),
    });
  }

  appLoadPolygonChart = appLoadPolygonChart.sort((a, b) => {
    if (a.point === undefined) {
      return 1;
    }

    if (b.point === undefined) {
      return -1;
    }

    return b.point - a.point;
  })

  const xAxisLength = width - 2 * margin;
  const yAxisLength = height - 2 * margin;

  const [maxX, minX] = findXRange(appLoadPolygonChart, xkey);
  const scaleX = scaleLinear()
    .domain([minX, maxX])
    .range([0, xAxisLength]);

  const [maxY, minY] = findYRange(appLoadPolygonChart, ykey);
  const scaleY = scaleLinear()
    .domain([maxY, minY])
    .range([0, yAxisLength]);

  const drawData = appLoadPolygonChart.map((item) => ({
    x: scaleX(moment.utc(item.timestamp)),
    y: scaleY(isMetricExists(item[ykey]) ? 0 : Math.max(maxY / 15, item[ykey])),
  }))

  const sparklineCanvas = useRef<HTMLCanvasElement>();

  useEffect(() => {
    if (!sparklineCanvas.current) {
      return;
    }

    const ctx = sparklineCanvas.current.getContext('2d');
    const drawBar = (barIndex: number, color: string, minHeight?: boolean): void => {
      ctx.fillStyle = color;
      const height = drawData[barIndex].y;
      const offsetX = drawData[barIndex].x;
      const width = drawData[barIndex - 1]
        ? drawData[barIndex].x - drawData[barIndex - 1].x
        : GRAPH_WIDTH / drawData.length;

      ctx.fillRect(
        offsetX,
        minHeight ? Math.min(BAR_HEIGHT - 2, height) : height,
        width,
        minHeight ? 30 : BAR_HEIGHT - height,
      );
    };

    const createTooltip = (columnNumber) => {
      const value = isMetricExists(appLoadPolygonChart[columnNumber][ykey])
        ? 0
        : appLoadPolygonChart[columnNumber][ykey];
      const dateToShow = moment(appLoadPolygonChart[columnNumber][xkey]).format('YYYY-MM-DD HH:mm:ss');

      // eslint-disable-next-line max-len
      const isTimeBased = metricName.endsWith('_time') || metricName.endsWith('_wait') || metricName === 'load';
      const load = humanize.transform(value, 'number');

      return !value ? `NA at ${dateToShow}` : `${load} ${isTimeBased ? '' : '/ sec'} at ${dateToShow}`;
    };

    sparklineCanvas.current.addEventListener('mousemove', (e) => {
      const columnNumber = Math.floor(e.offsetX / (GRAPH_WIDTH / (drawData.length - 1));

      setTooltip(createTooltip(columnNumber));

      updateGraphs(columnNumber);

      // ReactTooltip.show(sparklineCanvas);
      ctx.clearRect(0, 0, 300, 30);
      drawData.forEach((item, index) => {
        drawBar(index, color);
      });

      drawBar(columnNumber, 'white', true);
    });

    sparklineCanvas.current.addEventListener('mouseout', () => {
      updateGraphs(NaN);
      ctx.clearRect(0, 0, 300, 30);
      drawData.forEach((item, index) => {
        drawBar(index, color);
      });
    });

    // sparklineCanvas.current.addEventListener('mousedown', (e) => {
    //   const startPoint = e.offsetX;
    //   let endPoint = null;
    //
    //   e.stopPropagation();
    //   const mouseUpHandler = (e) => {
    //     ctx.clearRect(startPoint, 0, endPoint - startPoint, BAR_HEIGHT);
    //
    //     const start = Math.floor(startPoint / (GRAPH_WIDTH / drawData.length));
    //     const end = Math.floor(endPoint / (GRAPH_WIDTH / drawData.length));
    //
    //     const result = [appLoadPolygonChart[start], appLoadPolygonChart[end]]
    //       .sort((a, b) => b.point - a.point)
    //       .map((item) => item.timestamp);
    //
    //     console.log(result);
    //     e.stopPropagation();
    //     sparklineCanvas.current.removeEventListener('mouseup', mouseUpHandler);
    //     sparklineCanvas.current.removeEventListener('mouseup', mouseMove);
    //   };
    //   const mouseMove = (e) => {
    //     endPoint = e.offsetX;
    //
    //     // ctx.fillStyle = "rgba(150,150,150,0.3)";
    //     // const height = drawData[barIndex].y;
    //     // ctx.fillRect(startPoint, 0, endPoint - startPoint, BAR_HEIGHT);
    //
    //     e.stopPropagation();
    //   };
    //
    //   sparklineCanvas.current.addEventListener('mouseup', mouseUpHandler);
    //   sparklineCanvas.current.addEventListener('mousemove', mouseMove);
    // });
    ctx.clearRect(0, 0, 300, 30);
    drawData.forEach((item, index) => {
      drawBar(index, color);
    });

    const drawHighlighted = (e) => {
      ctx.clearRect(0, 0, 300, 30);
      drawData.forEach((item, index) => drawBar(index, color));

      if (!e.detail) {
        return;
      }

      const columnNumber = e.detail;

      drawBar(columnNumber, 'blue');
    };

    document.addEventListener('sync-graphs', drawHighlighted, false);

    // eslint-disable-next-line consistent-return
    return () => {
      document.removeEventListener('sync-graphs', drawHighlighted);
    };
  }, [sparklineCanvas]);

  return (
    <>
      <canvas
        ref={sparklineCanvas}
        style={{ background: 'transparent' }}
        width={GRAPH_WIDTH}
        height={BAR_HEIGHT}
        data-tip={tooltip}
        data-for="test"
      />
      <ReactTooltip
        data-qa="sparkline-tooltip"
        className="sparkline-tooltip"
        place="bottom"
        backgroundColor="#3274d9"
        arrowColor="#3274d9"
        id="test"
        getContent={() => {
          // console.log('tooltip', tooltip)
          return tooltip
        }}
      />
    </>
  );
};
