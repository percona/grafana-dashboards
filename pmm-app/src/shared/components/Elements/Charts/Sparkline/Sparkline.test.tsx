// Just a stub test
import React from 'react';
import { shallow } from 'enzyme';
import { Sparkline } from './Sparkline';

jest.mock('react-tooltip');

const MOCK_SPARKLINE = [
  {
    time_frame: 360,
    timestamp: '2020-02-13T15:41:00Z',
    metric: '',
  },
  {
    point: 1,
    time_frame: 360,
    timestamp: '2020-02-13T15:35:00Z',
    metric: '',
  },
  {
    point: 2,
    time_frame: 360,
    timestamp: '2020-02-13T15:29:00Z',
    metric: '',
  },
  {
    point: 3,
    time_frame: 360,
    timestamp: '2020-02-13T15:23:00Z',
    metric: '',
  },
  {
    point: 4,
    time_frame: 360,
    timestamp: '2020-02-13T15:17:00Z',
    metric: '',
  },
  {
    point: 5,
    time_frame: 360,
    timestamp: '2020-02-13T15:11:00Z',
    metric: '',
  },
  {
    point: 6,
    time_frame: 360,
    timestamp: '2020-02-13T15:05:00Z',
    metric: '',
  },
  {
    point: 7,
    time_frame: 360,
    timestamp: '2020-02-13T14:59:00Z',
    metric: '',
  },
  {
    point: 8,
    time_frame: 360,
    timestamp: '2020-02-13T14:53:00Z',
    metric: '',
  },
  {
    point: 9,
    time_frame: 360,
    timestamp: '2020-02-13T14:47:00Z',
    metric: '',
  },
  {
    point: 10,
    time_frame: 360,
    timestamp: '2020-02-13T14:41:00Z',
    metric: '',
  },
  {
    point: 11,
    time_frame: 360,
    timestamp: '2020-02-13T14:35:00Z',
    metric: '',
  },
  {
    point: 12,
    time_frame: 360,
    timestamp: '2020-02-13T14:29:00Z',
    metric: '',
  },
  {
    point: 13,
    time_frame: 360,
    timestamp: '2020-02-13T14:23:00Z',
    metric: '',
  },
  {
    point: 14,
    time_frame: 360,
    timestamp: '2020-02-13T14:17:00Z',
    metric: '',
  },
  {
    point: 15,
    time_frame: 360,
    timestamp: '2020-02-13T14:11:00Z',
    metric: '',
  },
  {
    point: 16,
    time_frame: 360,
    timestamp: '2020-02-13T14:05:00Z',
    metric: '',
  },
  {
    point: 17,
    time_frame: 360,
    timestamp: '2020-02-13T13:59:00Z',
    metric: '',
  },
  {
    point: 18,
    time_frame: 360,
    timestamp: '2020-02-13T13:53:00Z',
    metric: '',
  },
  {
    point: 19,
    time_frame: 360,
    timestamp: '2020-02-13T13:47:00Z',
    metric: '',
  },
  {
    point: 20,
    time_frame: 360,
    timestamp: '2020-02-13T13:41:00Z',
    metric: '',
  },
  {
    point: 21,
    time_frame: 360,
    timestamp: '2020-02-13T13:35:00Z',
    metric: '',
  },
  {
    point: 22,
    time_frame: 360,
    timestamp: '2020-02-13T13:29:00Z',
    metric: '',
  },
  {
    point: 23,
    time_frame: 360,
    timestamp: '2020-02-13T13:23:00Z',
    metric: '',
  },
  {
    point: 24,
    time_frame: 360,
    timestamp: '2020-02-13T13:17:00Z',
    metric: '',
  },
  {
    point: 25,
    time_frame: 360,
    timestamp: '2020-02-13T13:11:00Z',
    metric: '',
  },
  {
    point: 26,
    time_frame: 360,
    timestamp: '2020-02-13T13:05:00Z',
    metric: '',
  },
  {
    point: 27,
    time_frame: 360,
    timestamp: '2020-02-13T12:59:00Z',
    metric: '',
  },
  {
    point: 28,
    time_frame: 360,
    timestamp: '2020-02-13T12:53:00Z',
    metric: '',
  },
  {
    point: 29,
    time_frame: 360,
    timestamp: '2020-02-13T12:47:00Z',
    metric: '',
  },
  {
    point: 30,
    time_frame: 360,
    timestamp: '2020-02-13T12:41:00Z',
    metric: '',
  },
  {
    point: 31,
    time_frame: 360,
    timestamp: '2020-02-13T12:35:00Z',
    metric: '',
  },
  {
    point: 32,
    time_frame: 360,
    timestamp: '2020-02-13T12:29:00Z',
    metric: '',
  },
  {
    point: 33,
    time_frame: 360,
    timestamp: '2020-02-13T12:23:00Z',
    metric: '',
  },
  {
    point: 34,
    time_frame: 360,
    timestamp: '2020-02-13T12:17:00Z',
    metric: '',
  },
  {
    point: 35,
    time_frame: 360,
    timestamp: '2020-02-13T12:11:00Z',
    metric: '',
  },
  {
    point: 36,
    time_frame: 360,
    timestamp: '2020-02-13T12:05:00Z',
    metric: '',
  },
  {
    point: 37,
    time_frame: 360,
    timestamp: '2020-02-13T11:59:00Z',
    metric: '',
  },
  {
    point: 38,
    time_frame: 360,
    timestamp: '2020-02-13T11:53:00Z',
    metric: '',
  },
  {
    point: 39,
    time_frame: 360,
    timestamp: '2020-02-13T11:47:00Z',
    metric: '',
  },
  {
    point: 40,
    time_frame: 360,
    timestamp: '2020-02-13T11:41:00Z',
    metric: '',
  },
  {
    point: 41,
    time_frame: 360,
    timestamp: '2020-02-13T11:35:00Z',
    metric: '',
  },
  {
    point: 42,
    time_frame: 360,
    timestamp: '2020-02-13T11:29:00Z',
    metric: '',
  },
  {
    point: 43,
    time_frame: 360,
    timestamp: '2020-02-13T11:23:00Z',
    metric: '',
  },
  {
    point: 44,
    time_frame: 360,
    timestamp: '2020-02-13T11:17:00Z',
    metric: '',
  },
  {
    point: 45,
    time_frame: 360,
    timestamp: '2020-02-13T11:11:00Z',
    metric: '',
  },
  {
    point: 46,
    time_frame: 360,
    timestamp: '2020-02-13T11:05:00Z',
    metric: '',
  },
  {
    point: 47,
    time_frame: 360,
    timestamp: '2020-02-13T10:59:00Z',
    metric: '',
  },
  {
    point: 48,
    time_frame: 360,
    timestamp: '2020-02-13T10:53:00Z',
    metric: '',
  },
  {
    point: 49,
    time_frame: 360,
    timestamp: '2020-02-13T10:47:00Z',
    metric: '',
  },
  {
    point: 50,
    time_frame: 360,
    timestamp: '2020-02-13T10:41:00Z',
    metric: 0.011111111,
  },
  {
    point: 51,
    time_frame: 360,
    timestamp: '2020-02-13T10:35:00Z',
    metric: 0.033333335,
  },
  {
    point: 52,
    time_frame: 360,
    timestamp: '2020-02-13T10:29:00Z',
    metric: 0.044444446,
  },
  {
    point: 53,
    time_frame: 360,
    timestamp: '2020-02-13T10:23:00Z',
    metric: '',
  },
  {
    point: 54,
    time_frame: 360,
    timestamp: '2020-02-13T10:17:00Z',
    metric: '',
  },
  {
    point: 55,
    time_frame: 360,
    timestamp: '2020-02-13T10:11:00Z',
    metric: '',
  },
  {
    point: 56,
    time_frame: 360,
    timestamp: '2020-02-13T10:05:00Z',
    metric: '',
  },
  {
    point: 57,
    time_frame: 360,
    timestamp: '2020-02-13T09:59:00Z',
    metric: '',
  },
  {
    point: 58,
    time_frame: 360,
    timestamp: '2020-02-13T09:53:00Z',
    metric: '',
  },
  {
    point: 59,
    time_frame: 360,
    timestamp: '2020-02-13T09:47:00Z',
    metric: '',
  },
  {
    point: 60,
    time_frame: 360,
    timestamp: '2020-02-13T09:41:00Z',
    metric: '',
  },
  {
    point: 61,
    time_frame: 360,
    timestamp: '2020-02-13T09:35:00Z',
    metric: '',
  },
  {
    point: 62,
    time_frame: 360,
    timestamp: '2020-02-13T09:29:00Z',
    metric: '',
  },
  {
    point: 63,
    time_frame: 360,
    timestamp: '2020-02-13T09:23:00Z',
    metric: '',
  },
  {
    point: 64,
    time_frame: 360,
    timestamp: '2020-02-13T09:17:00Z',
    metric: '',
  },
  {
    point: 65,
    time_frame: 360,
    timestamp: '2020-02-13T09:11:00Z',
    metric: '',
  },
  {
    point: 66,
    time_frame: 360,
    timestamp: '2020-02-13T09:05:00Z',
    metric: '',
  },
  {
    point: 67,
    time_frame: 360,
    timestamp: '2020-02-13T08:59:00Z',
    metric: '',
  },
  {
    point: 68,
    time_frame: 360,
    timestamp: '2020-02-13T08:53:00Z',
    metric: '',
  },
  {
    point: 69,
    time_frame: 360,
    timestamp: '2020-02-13T08:47:00Z',
    metric: '',
  },
  {
    point: 70,
    time_frame: 360,
    timestamp: '2020-02-13T08:41:00Z',
    metric: '',
  },
  {
    point: 71,
    time_frame: 360,
    timestamp: '2020-02-13T08:35:00Z',
    metric: '',
  },
  {
    point: 72,
    time_frame: 360,
    timestamp: '2020-02-13T08:29:00Z',
    metric: '',
  },
  {
    point: 73,
    time_frame: 360,
    timestamp: '2020-02-13T08:23:00Z',
    metric: '',
  },
  {
    point: 74,
    time_frame: 360,
    timestamp: '2020-02-13T08:17:00Z',
    metric: '',
  },
  {
    point: 75,
    time_frame: 360,
    timestamp: '2020-02-13T08:11:00Z',
    metric: '',
  },
  {
    point: 76,
    time_frame: 360,
    timestamp: '2020-02-13T08:05:00Z',
    metric: '',
  },
  {
    point: 77,
    time_frame: 360,
    timestamp: '2020-02-13T07:59:00Z',
    metric: '',
  },
  {
    point: 78,
    time_frame: 360,
    timestamp: '2020-02-13T07:53:00Z',
    metric: '',
  },
  {
    point: 79,
    time_frame: 360,
    timestamp: '2020-02-13T07:47:00Z',
    metric: '',
  },
  {
    point: 80,
    time_frame: 360,
    timestamp: '2020-02-13T07:41:00Z',
    metric: '',
  },
  {
    point: 81,
    time_frame: 360,
    timestamp: '2020-02-13T07:35:00Z',
    metric: '',
  },
  {
    point: 82,
    time_frame: 360,
    timestamp: '2020-02-13T07:29:00Z',
    metric: '',
  },
  {
    point: 83,
    time_frame: 360,
    timestamp: '2020-02-13T07:23:00Z',
    metric: '',
  },
  {
    point: 84,
    time_frame: 360,
    timestamp: '2020-02-13T07:17:00Z',
    metric: '',
  },
  {
    point: 85,
    time_frame: 360,
    timestamp: '2020-02-13T07:11:00Z',
    metric: '',
  },
  {
    point: 86,
    time_frame: 360,
    timestamp: '2020-02-13T07:05:00Z',
    metric: '',
  },
  {
    point: 87,
    time_frame: 360,
    timestamp: '2020-02-13T06:59:00Z',
    metric: '',
  },
  {
    point: 88,
    time_frame: 360,
    timestamp: '2020-02-13T06:53:00Z',
    metric: '',
  },
  {
    point: 89,
    time_frame: 360,
    timestamp: '2020-02-13T06:47:00Z',
    metric: '',
  },
  {
    point: 90,
    time_frame: 360,
    timestamp: '2020-02-13T06:41:00Z',
    metric: '',
  },
  {
    point: 91,
    time_frame: 360,
    timestamp: '2020-02-13T06:35:00Z',
    metric: '',
  },
  {
    point: 92,
    time_frame: 360,
    timestamp: '2020-02-13T06:29:00Z',
    metric: '',
  },
  {
    point: 93,
    time_frame: 360,
    timestamp: '2020-02-13T06:23:00Z',
    metric: '',
  },
  {
    point: 94,
    time_frame: 360,
    timestamp: '2020-02-13T06:17:00Z',
    metric: '',
  },
  {
    point: 95,
    time_frame: 360,
    timestamp: '2020-02-13T06:11:00Z',
    metric: '',
  },
  {
    point: 96,
    time_frame: 360,
    timestamp: '2020-02-13T06:05:00Z',
    metric: '',
  },
  {
    point: 97,
    time_frame: 360,
    timestamp: '2020-02-13T05:59:00Z',
    metric: '',
  },
  {
    point: 98,
    time_frame: 360,
    timestamp: '2020-02-13T05:53:00Z',
    metric: '',
  },
  {
    point: 99,
    time_frame: 360,
    timestamp: '2020-02-13T05:47:00Z',
    metric: '',
  },
  {
    point: 100,
    time_frame: 360,
    timestamp: '2020-02-13T05:41:00Z',
    metric: '',
  },
  {
    point: 101,
    time_frame: 360,
    timestamp: '2020-02-13T05:35:00Z',
    metric: '',
  },
  {
    point: 102,
    time_frame: 360,
    timestamp: '2020-02-13T05:29:00Z',
    metric: '',
  },
  {
    point: 103,
    time_frame: 360,
    timestamp: '2020-02-13T05:23:00Z',
    metric: '',
  },
  {
    point: 104,
    time_frame: 360,
    timestamp: '2020-02-13T05:17:00Z',
    metric: '',
  },
  {
    point: 105,
    time_frame: 360,
    timestamp: '2020-02-13T05:11:00Z',
    metric: '',
  },
  {
    point: 106,
    time_frame: 360,
    timestamp: '2020-02-13T05:05:00Z',
    metric: '',
  },
  {
    point: 107,
    time_frame: 360,
    timestamp: '2020-02-13T04:59:00Z',
    metric: '',
  },
  {
    point: 108,
    time_frame: 360,
    timestamp: '2020-02-13T04:53:00Z',
    metric: '',
  },
  {
    point: 109,
    time_frame: 360,
    timestamp: '2020-02-13T04:47:00Z',
    metric: '',
  },
  {
    point: 110,
    time_frame: 360,
    timestamp: '2020-02-13T04:41:00Z',
    metric: '',
  },
  {
    point: 111,
    time_frame: 360,
    timestamp: '2020-02-13T04:35:00Z',
    metric: '',
  },
  {
    point: 112,
    time_frame: 360,
    timestamp: '2020-02-13T04:29:00Z',
    metric: '',
  },
  {
    point: 113,
    time_frame: 360,
    timestamp: '2020-02-13T04:23:00Z',
    metric: '',
  },
  {
    point: 114,
    time_frame: 360,
    timestamp: '2020-02-13T04:17:00Z',
    metric: '',
  },
  {
    point: 115,
    time_frame: 360,
    timestamp: '2020-02-13T04:11:00Z',
    metric: '',
  },
  {
    point: 116,
    time_frame: 360,
    timestamp: '2020-02-13T04:05:00Z',
    metric: '',
  },
  {
    point: 117,
    time_frame: 360,
    timestamp: '2020-02-13T03:59:00Z',
    metric: '',
  },
  {
    point: 118,
    time_frame: 360,
    timestamp: '2020-02-13T03:53:00Z',
    metric: '',
  },
  {
    point: 119,
    time_frame: 360,
    timestamp: '2020-02-13T03:47:00Z',
    metric: '',
  },
];

describe('Polygon chart test', () => {
  it('Renders correct with right props', () => {
    const polygonChartProps = {
      data: MOCK_SPARKLINE,
      width: 150,
      ykey: 'metric',
    };

    const component = shallow(<Sparkline {...polygonChartProps} />);

    expect(component).toMatchSnapshot();
  });
});
