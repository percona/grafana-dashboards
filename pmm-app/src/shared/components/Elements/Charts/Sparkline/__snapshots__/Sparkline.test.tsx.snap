// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Polygon chart test Renders correct with right props 1`] = `
<Fragment>
  <canvas
    data-for="test"
    data-tip=""
    height={30}
    style={
      Object {
        "background": "transparent",
      }
    }
    width={300}
  />
  <ReactTooltip
    arrowColor="#3274d9"
    backgroundColor="#3274d9"
    className="sparkline-tooltip"
    clickable={false}
    data-qa="sparkline-tooltip"
    getContent={[Function]}
    id="test"
    insecure={true}
    place="bottom"
    resizeHide={true}
    wrapper="div"
  />
</Fragment>
`;
